pkg/controller/planexecution/planexecution_controller.go

```diff
@@ -228,6 +227,11 @@ func (r *ReconcilePlanExecution) Reconcile(request reconcile.Request) (reconcile
 		return reconcile.Result{}, err
 	}
 
+	if instance.Status.ActivePlan.Name != planExecution.Name || instance.Status.ActivePlan.Namespace != planExecution.Namespace {
```

Comment:
this is needed because otherwise the plan runs until the end but then cannot update instance because that one was modified in between

Reply:
in general I feel like this safeguard is correct. If we're reconciling PE that is not current active plan, we probably should not execute anything

Reply:
Couldn't this be executed right after the PE is created, but before the instance controller calls `addActivePlanReference`?

Reply:
hmm good point, I guess it could. To overcome even that case I guess adding reconcile trigger after linking activeplan could be a solution? But I still think this safeguard is correct, because we should really execute only active plan, not every plan that changed for example...

Reply:
oh wait, yeah, that should be fine, I think the PE controller should also watch for instance update event and trigger reconciliation as it does for deployments etc. Would you agree? Looking at the implementation right now I am not 100% that is happening and that it was intentional but if we agree on that I can push in that direction